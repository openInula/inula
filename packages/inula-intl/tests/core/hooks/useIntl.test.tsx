/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2023-2023. All rights reserved.
 */

import * as React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider, useIntl } from '../../../index';

const FunctionComponent = ({ spy }: { spy?: Function }) => {
  const { i18n } = useIntl();
  spy!(i18n.locale);
  return null;
};

const FC = () => {
  const i18n = useIntl();
  return i18n.formatNumber(10000, { style: 'currency', currency: 'USD' }) as any;
};

describe('useIntl() hooks', () => {
  it('throws when <IntlProvider> is missing from ancestry', () => {
    // So it doesn't spam the console
    jest.spyOn(console, 'error').mockImplementation(() => {});
    expect(() => render(<FunctionComponent />)).toThrow('I18n object is not found!');
  });

  it('hooks onto the intl context', () => {
    const spy = jest.fn();
    render(
      <IntlProvider locale="en">
        <FunctionComponent spy={spy} />
      </IntlProvider>
    );
    expect(spy).toHaveBeenCalledWith('en');
  });

  it('should work when switching locale on provider', () => {
    const { rerender, getByTestId } = render(
      <IntlProvider locale="en">
        <span data-testid="comp">
          <FC />
        </span>
      </IntlProvider>
    );
    expect(getByTestId('comp')).toMatchSnapshot();
    rerender(
      <IntlProvider locale="es">
        <span data-testid="comp">
          <FC />
        </span>
      </IntlProvider>
    );
    expect(getByTestId('comp')).toMatchSnapshot();
    rerender(
      <IntlProvider locale="en">
        <span data-testid="comp">
          <FC />
        </span>
      </IntlProvider>
    );

    expect(getByTestId('comp')).toMatchSnapshot();
  });
});
