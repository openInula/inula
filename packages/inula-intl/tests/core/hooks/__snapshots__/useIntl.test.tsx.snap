// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useIntl() hooks should work when switching locale on provider 1`] = `"<span>$10,000.00</span>"`;

exports[`useIntl() hooks should work when switching locale on provider 2`] = `"<span>10.000,00&nbsp;US$</span>"`;

exports[`useIntl() hooks should work when switching locale on provider 3`] = `"<span>$10,000.00</span>"`;
